internal/backend/sessions/sessiondata/data.go

```
@@ -58,32 +56,22 @@ func downloadBuild(ctx context.Context, buildID sdktypes.BuildID, builds sdkserv
 func Get(ctx context.Context, svcs *sessionsvcs.Svcs, session sdktypes.Session) (*Data, error) {
 	var err error
 
-	data := Data{
-		Session: session,
-	}
+	data := Data{Session: session}
 
 	// TODO(ENG-207): Consider doing all retrievals using one big happy join.
 
-	if envID := data.Session.EnvID(); envID.IsValid() {
-		if data.Env, err = retrieve(ctx, envID, svcs.Envs.GetByID); err != nil {
-			return nil, err
-		}
-
-		if data.ProjectID = data.Env.ProjectID(); !data.ProjectID.IsValid() {
-			return nil, fmt.Errorf("sessions can only run on projects")
-		}
-
-		cfilter := sdkservices.ListConnectionsFilter{ProjectID: data.ProjectID}
+	if pid := session.ProjectID(); pid.IsValid() {
```

i think env wasn't required before
but can we have a session without a valid projectID ?
maybe it should return an error if the projectID is invalid on the session ?

---

so for session tests there is no project id. theoretically one might launch sessions without a project.

---

it just make the system more complicated for stuff we don't really need, nor there are requirements for it
I think error it better here

---

i think it is useful, not just for testing. will limit it with rbac later.